framework/include/vectorpostprocessors/LineMaterialSamplerBase.h

```diff
@@ -179,6 +184,7 @@ LineMaterialSamplerBase<T>::execute()
       addSample(_q_point[qp], qp_proj_dist_along_line, values);
     }
   }
+  _fe_problem.clearActiveMaterialProperties(_tid);
```

Comment:
Right this is **outside** the elemental loop. This is a `GeneralUserObject` which is why it doesn't automatically know about material dependencies. These cases are rare but they aren't zero. Looking for a better solution is a bigger problem that will take some redesign and if we do come up with a solution, it'll go in several places (not just here).